src/user.rs

```diff
@@ -78,20 +75,53 @@ pub fn handle_claim_rewards<S: Storage, A: Api, Q: Querier>(
     })
 }
 
-pub fn handle_increase_balance<S: Storage, A: Api, Q: Querier>(
+/*
+    Encode and prepare message to perfom a TransferFrom cw-20 LOTA contract in order to transfer
+    staker LOTA to staking contract funds are locked as a custodian
+*/
+fn encode_msg_execute(msg: QueryMsg, address: HumanAddr) -> StdResult<CosmosMsg> {
+    Ok(WasmMsg::Execute {
+        contract_addr: address,
+        msg: to_binary(&msg)?,
+        send: vec![],
+    }
+        .into())
+}
+pub fn handle_bond<S: Storage, A: Api, Q: Querier>(
     deps: &mut Extern<S, A, Q>,
     env: Env,
-    address: HumanAddr,
     amount: Uint128,
-) -> StdResult<HandleResponse<TerraMsgWrapper>> {
+) -> StdResult<HandleResponse> {
     let config = read_config(&deps.storage)?;
-    let address_raw = deps.api.canonical_address(&address)?;
+    let address_raw = deps.api.canonical_address(&env.message.sender)?;
     let sender = env.message.sender;
-
-    // Check sender is token contract
-    if sender != deps.api.human_address(&config.cw20_token_contract)? {
-        return Err(StdError::unauthorized());
+    /*
+        TODO: Add safe_lock
+     */
+    /*if state.safe_lock {
+        return Err(StdError::generic_err(
+            "Contract deactivated for update or/and preventing security issue",
+        ));
+    }*/
+
+    if !env.message.sent_funds.is_empty() {
+        return Err(StdError::generic_err("Do not send funds with stake"));
     }
+    if amount.is_zero() {
+        return Err(StdError::generic_err("Amount required"));
+    }
+    // Prepare msg to send
+    let prepare_msg = QueryMsg::TransferFrom {
```

Comment:
Import and use HandleMsg::TransferFrom from cosmwasm-plus v0.2.3 https://github.com/CosmWasm/cosmwasm-plus/blob/v0.2.3/contracts/cw20-base/src/msg.rs#L101